controllers/ie_controller_test.go

```diff
@@ -0,0 +1,177 @@
+package controllers
```

Comment:
Putting the test in the same package as source file is standard go practice.

Code Loader is pre-existing, but I can change the file name.